### PR TITLE
SD-8428: Update Github actions tooling

### PR DESCRIPTION
**Pull request by JureMalovrh**
Take new tooling from scaffold to be up to date

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,45 +1,60 @@
...
     steps:
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      - uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+      - run: go mod download
+      - run: go build -v .
+        # unused until we remove the old code for 0.6 version
+#      - name: Run linters
```
Comment (zigomir) [inline .github/workflows/test.yml:30]: Why skip linters?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,45 +1,60 @@
...
+      # Temporarily download Terraform 1.8 prerelease for function documentation support.
+      # When Terraform 1.8.0 final is released, this can be removed.
+      - uses: hashicorp/setup-terraform@651471c36a6092792c552e8b1bef71e592b462d8 # v3.1.1
+        with:
+          terraform_version: '1.8.0-alpha20240216'
+          terraform_wrapper: false
+      - run: go generate ./...
+      - name: git diff
+        run: |
+          git diff --compact-summary --exit-code || \
```
Comment (zigomir) [inline .github/workflows/test.yml:52]: I see this command is failing atm: https://github.com/sleuth-io/terraform-provider-sleuth/actions/runs/9158897361/job/25178177667?pr=168

**Review by zigomir — APPROVED** [review]
🤷

Funny we depend on this `crazy-max` though :D

File: .github/workflows/test.yml (hunk 1 above)
Comment (JureMalovrh) [reply .github/workflows/test.yml:30]: Becasue there are too many problems from old versions - in 0.6 we will remove old code, thus also removing the issues and then linters can get re-enabled.

File: .github/workflows/test.yml (hunk 2 above)
Comment (JureMalovrh) [reply .github/workflows/test.yml:52]: yes, it will start working in next PR with documentation update

**Pull request merged**